src/v/storage/disk_log_impl.cc

```diff
@@ -1565,15 +1587,26 @@ ss::future<> disk_log_impl::apply_segment_ms() {
         co_return;
     }
 
+    auto and_then = [](std::optional<int64_t>& val)
+      -> std::optional<std::chrono::milliseconds> {
+        if (val) {
+            return std::make_optional(
+              static_cast<std::chrono::milliseconds>(*val));
+        }
+        return std::nullopt;
+    };
+
     auto& local_config = config::shard_local_cfg();
+    std::optional<int64_t> min_opt, max_opt;
+    config::set_min_max<int64_t>("log_segment_ms", min_opt, max_opt);
     // clamp user provided value with (hopefully sane) server min and max
-    // values, this should protect against overflow UB
+    // values, this should protect against overflow UB.
     if (
       first_write_ts.value()
         + std::clamp(
           seg_ms.value(),
-          local_config.log_segment_ms_min(),
-          local_config.log_segment_ms_max())
```

Comment:
@andijcr I am under the impression that the broker is meant to always apply the existing clamps for segment_ms and segment_size. Do I have that correct?

Reply:
yes, to be able to control at runtime the behavior of segment rolling

Reply:
and for segment.ms, the benefit of doing it like this (instead of saving the clamped value at segment creation) is to be able to intervene on the current segment, instead of waiting for rolling to occur naturally with the previous value